modular_skyrat/master_files/code/game/objects/structures/towel_bins.dm

```diff
@@ -0,0 +1,146 @@
+	icon_state = "linenbin-empty"
+	anchored = FALSE
+
+
+/obj/structure/towel_bin/examine(mob/user)
+	. = ..()
+	if(amount <= 0)
+		. += "There are no towels in the bin."
+	else
+		. += "There [amount == 1 ? "is one towel" : "are [amount] towels"] in the bin."
+
+
+/obj/structure/towel_bin/update_icon_state()
+	switch(amount)
+		if(0)
+			icon_state = "linenbin-empty"
+		if(1 to 5)
+			icon_state = "linenbin-half"
+		else
+			icon_state = "linenbin-full"
+	return ..()
+
+
+/obj/structure/towel_bin/fire_act(exposed_temperature, exposed_volume)
+	if(amount)
+		amount = 0
+		update_appearance()
+
+	return ..()
+
+
+/obj/structure/towel_bin/screwdriver_act(mob/living/user, obj/item/tool)
+	if(flags_1 & NODECONSTRUCT_1)
+		return FALSE
+
+	if(amount)
+		to_chat(user, span_warning("[src] must be empty first!"))
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+
+	if(tool.use_tool(src, user, 0.5 SECONDS, volume = 50))
+		to_chat(user, span_notice("You disassemble [src]."))
+		new /obj/item/stack/rods(loc, 2)
+		qdel(src)
+		return TOOL_ACT_TOOLTYPE_SUCCESS
+
+
+/obj/structure/towel_bin/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+	default_unfasten_wrench(user, tool, time = 0.5 SECONDS)
+	return TOOL_ACT_TOOLTYPE_SUCCESS
+
+
+/obj/structure/towel_bin/attackby(obj/item/attacking_item, mob/user, params)
+	if(istype(attacking_item, /obj/item/towel))
+		if(!user.transferItemToLoc(attacking_item, src))
+			return
+		LAZYADD(towels, attacking_item)
+		amount++
+		to_chat(user, span_notice("You put [attacking_item] in [src]."))
+		update_appearance()
+
+	else if(amount && !hidden && attacking_item.w_class < WEIGHT_CLASS_BULKY) //make sure there's sheets to hide it among, make sure nothing else is hidden in there.
+		if(!user.transferItemToLoc(attacking_item, src))
+			to_chat(user, span_warning("[attacking_item] is stuck to your hand, you cannot hide it among the sheets!"))
+			return
+		hidden = attacking_item
+		to_chat(user, span_notice("You hide [attacking_item] among the sheets."))
+
+
+/obj/structure/towel_bin/attack_paw(mob/user, list/modifiers)
+	return attack_hand(user, modifiers)
+
+
+/obj/structure/towel_bin/attack_hand(mob/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+
+	if(isliving(user))
+		var/mob/living/living_user = user
+		if(!(living_user.mobility_flags & MOBILITY_PICKUP))
+			return
+
+	take_towel_out(user)
+
+
+/obj/structure/towel_bin/attack_tk(mob/user)
+	take_towel_out(user, tk = TRUE)
+
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+
+/**
+ * Helper proc for taking a towel out of the bin, to reduce code repetitions.
+ * Intended to only be called by `attack_hand()` and `attack_tk()`.
+ *
+ * Arguments:
+ * * user - Mob that's trying to take a towel out.
+ * * tk - Is the user trying to do this using telekinesis? Defaults to `FALSE`.
+ */
+/obj/structure/towel_bin/proc/take_towel_out(mob/user, tk = FALSE)
+	if(amount <= 0)
+		to_chat(user, span_warning("You can't figure out how to take a towel out of [src] when it doesn't contain any!"))
```

Comment:
```suggestion
		balloon_alert(user, "bin is empty!")
```
All the homies hate to_chat failiures
Bitches LOVE balloon alerts

Reply:
This is just code that was copied out of the bedsheet bin code, I think.

Reply:
😭 another thing to fix on TG...... nooo....